tests/integration/test_bayesian_optimization.py

```diff
@@ -46,55 +48,65 @@
+from trieste.types import State, TensorType
 
 
 @random_seed
 @pytest.mark.parametrize(
     "num_steps, acquisition_rule",
-    [
-        (20, EfficientGlobalOptimization()),
-        (25, EfficientGlobalOptimization(AugmentedExpectedImprovement().using(OBJECTIVE))),
-        (
-            15,
-            EfficientGlobalOptimization(
-                MinValueEntropySearch(BRANIN_SEARCH_SPACE, num_fourier_features=1000).using(
-                    OBJECTIVE
-                )
+    cast(
+        List[
+            Union[
+                AcquisitionRule[TensorType, Box],
+                AcquisitionRule[State[TensorType, TrustRegion.State], Box],
+            ]
+        ],
+        [
+            (20, EfficientGlobalOptimization()),
+            (25, EfficientGlobalOptimization(AugmentedExpectedImprovement().using(OBJECTIVE))),
+            (
+                15,
+                EfficientGlobalOptimization(
+                    MinValueEntropySearch(BRANIN_SEARCH_SPACE, num_fourier_features=1000).using(
+                        OBJECTIVE
+                    )
+                ),
             ),
-        ),
-        (
-            10,
-            EfficientGlobalOptimization(
-                BatchMonteCarloExpectedImprovement(sample_size=500).using(OBJECTIVE),
-                num_query_points=3,
+            (
+                10,
+                EfficientGlobalOptimization(
+                    BatchMonteCarloExpectedImprovement(sample_size=500).using(OBJECTIVE),
+                    num_query_points=3,
+                ),
             ),
-        ),
-        (
-            10,
-            EfficientGlobalOptimization(
-                LocalPenalizationAcquisitionFunction(
-                    BRANIN_SEARCH_SPACE,
-                ).using(OBJECTIVE),
-                num_query_points=3,
+            (
+                10,
+                EfficientGlobalOptimization(
+                    LocalPenalizationAcquisitionFunction(
+                        BRANIN_SEARCH_SPACE,
+                    ).using(OBJECTIVE),
+                    num_query_points=3,
+                ),
             ),
-        ),
-        (
-            10,
-            EfficientGlobalOptimization(
-                GIBBON(
-                    BRANIN_SEARCH_SPACE,
-                ).using(OBJECTIVE),
-                num_query_points=2,
+            (
+                10,
+                EfficientGlobalOptimization(
+                    GIBBON(
+                        BRANIN_SEARCH_SPACE,
+                    ).using(OBJECTIVE),
+                    num_query_points=2,
+                ),
             ),
-        ),
-        (15, TrustRegion()),
-        (10, DiscreteThompsonSampling(500, 3)),
-        (10, DiscreteThompsonSampling(500, 3, num_fourier_features=1000)),
-    ],
+            (15, TrustRegion()),
+            (10, DiscreteThompsonSampling(500, 3)),
+            (10, DiscreteThompsonSampling(500, 3, num_fourier_features=1000)),
+        ],
+    ),
```

Comment:
Could we have an integration test of Trust Region combined with a more complicated acq? e.g MES?

Reply:
in addition to the one for trust region with ego or instead of?

Reply:
aswell

Reply:
added mes